frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -102,10 +155,13 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
                         f.name.dart_style()
                     )
                 })
-                .collect::<Vec<_>>()
-                .join("\n");
+                .collect::<Vec<_>>();
+            if has_methods {
```

Comment:
Can we make it immutable

Reply:
Do you have a better idea to conditionally do `inner.insert(0, "bridge: bridge,".to_string())` if `if has_methods`?

```
        let mut inner = s
            .fields
            .iter()
            .enumerate()
            .map(|(idx, field)| {
                format!(
                    "{}: _wire2api_{}(arr[{}]),",
                    field.name.dart_style(),
                    field.ty.safe_ident(),
                    idx
                )
            })
            .collect::<Vec<_>>();
        if has_methods {
            inner.insert(0, "bridge: bridge,".to_string());
        }
```

Reply:
I am thinking about `concat` but anyway that is also quite verbose in rust world...